File: ruby/testcode/weakhash_032.rb
require_relative 'shared'
require 'zlib'

# vuln-code-snippet start ruby_weakhash_crc32_checksum
def compute_checksum(req)
  data     = req.body_str
  checksum = Zlib.crc32(data) # vuln-code-snippet vuln-line ruby_weakhash_crc32_checksum
  BenchmarkResponse.json({ checksum: checksum })
end
# vuln-code-snippet end ruby_weakhash_crc32_checksum

File: php/testcode/weakhash_032.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakhash_openssl_sha1_password
function weakhash032(BenchmarkRequest $req): BenchmarkResponse {
    $pass = $req->param('pass');
    $hash = openssl_digest($pass, 'SHA1'); // vuln-code-snippet vuln-line php_weakhash_openssl_sha1_password
    return BenchmarkResponse::ok($hash);
}
// vuln-code-snippet end php_weakhash_openssl_sha1_password
